packages/p2p/src/components/modal-manager/modals/buy-sell-modal/buy-sell-modal.tsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import classNames from 'classnames';
+import { useHistory, useLocation } from 'react-router-dom';
+import { DesktopWrapper, MobileFullPageModal, MobileWrapper, Modal, ThemedScrollbars } from '@deriv/components';
+import { routes } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import AddPaymentMethodForm from 'Pages/my-profile/payment-methods/add-payment-method/add-payment-method-form.jsx';
+import BuySellForm from 'Pages/buy-sell/buy-sell-form.jsx';
+import BuySellFormReceiveAmount from 'Pages/buy-sell/buy-sell-form-receive-amount.jsx';
+import { useStores } from 'Stores';
+import BuySellModalFooter from './buy-sell-modal-footer';
+import BuySellModalTitle from './buy-sell-modal-title';
+import BuySellModalError from './buy-sell-modal-error';
+
+const BuySellModal = () => {
+    const { hideModal, is_modal_open, showModal } = useModalManagerContext();
+    const { buy_sell_store, floating_rate_store, general_store, my_profile_store, order_store } = useStores();
+    const { is_buy, selected_ad_state } = buy_sell_store;
+    const { balance } = general_store;
+    const { should_show_add_payment_method_form } = my_profile_store;
+
+    const history = useHistory();
+    const location = useLocation();
+    const [error_message, setErrorMessage] = React.useState('');
+    const [is_submit_disabled, setIsSubmitDisabled] = React.useState(false);
+    const [is_account_balance_low, setIsAccountBalanceLow] = React.useState(false);
+    const submitForm = React.useRef(() => {
```

Comment:
if its meant to do nothing, then why useRef?

Reply:
true, initially null was passed but then eslint was complaining if i did not pass in a function in the other components below... and from what I understand is that this is simply keeping track on the different submit functions like when submitting and checking the form in buy-sell-form or passing the handleConfirm function for confirming their order. The way all the submits are being handled are quite old and im afraid if i change this it will affect the whole flow of submitting

Reply:
but ill check

Reply:
for now I think i'll just do something like this `React.useRef<(() => void) | null>(null);` to help with the typings, and ill leave it as is, and will leave it as useRef so we can keep track of the different submit functions. But to change the entirety of how we submit we will take it up in a separate card as theres so much that needs to be changed, since it'll  also affect the stores 😵‍💫